src/client/windshaft.js

```diff
@@ -401,9 +401,7 @@ export default class Windshaft {
 
         const idProperty = 'cartodb_id';
 
-        const metadata = new Metadata({ properties, featureCount, sample: stats.sample, geomType, isAggregated: aggregation.mvt, idProperty });
-        metadata.setCodecs();
-        return metadata;
+        return new Metadata({ properties, featureCount, sample: stats.sample, geomType, isAggregated: aggregation.mvt, idProperty });
```

Comment:
Should this same change be applied to GeoJSON?

Reply:
If base `Metadata` is not the place anymore to properly `setCodecs`, maybe we could 'defend' from unwanted uses creating there something like:

```js
    setCodecs() {
        throw new CartoRuntimeError(`You must call 'setCodecs' once you have determined the proper subclass`);
    }
```

Does it make sense to you?

Reply:
I might not be understanding your comment about GeoJSON. We are already doing something similar in the GeoJSONN source, see https://github.com/CartoDB/carto-vl/blob/a2fdec46c9707a3b7b3ab0cb584a256d37ebe9cd/src/sources/GeoJSON.js#L159-L160

Reply:
Fair point about the deffensive code in the base metadata class. I will add that.

Reply:
Regarding to GeoJSON, I thought both lines would rise this very same 'deffensive error', but I have just seen that Metadata there is already a GeoJSONMetadata, so forget about it.